tests/test_coroutines.py

```diff
@@ -0,0 +1,80 @@
+import asyncio
+import gc
+from typing import Any
+from unittest.mock import Mock
+
+import pytest
+
+from psygnal import _async
+from psygnal._weak_callback import WeakCallback, weak_callback
+
+
+@pytest.mark.parametrize(
+    "type_",
+    [
+        "coroutinefunc",
+        "weak_coroutinefunc",
+        "coroutinemethod",
+    ],
+)
+@pytest.mark.asyncio
+async def test_slot_types(type_: str, capsys: Any) -> None:
+    backend = _async.set_async_backend("asyncio")
+    assert backend is _async.get_async_backend() is not None
+    while not backend.running:
+        await asyncio.sleep(0)
+
+    mock = Mock()
+    final_mock = Mock()
+
+    obj: Any
+    if type_ in {"coroutinefunc", "weak_coroutinefunc"}:
+
+        async def obj(x: int) -> int:
+            mock(x)
+            return x
+
+        cb = weak_callback(
+            obj, strong_func=(type_ == "coroutinefunc"), finalize=final_mock
+        )
+    elif type_ == "coroutinemethod":
+
+        class MyObj:
+            async def coroutine_method(self, x: int) -> int:
+                mock(x)
+                return x
+
+        obj = MyObj()
+        cb = weak_callback(obj.coroutine_method, finalize=final_mock)
+
+    assert isinstance(cb, WeakCallback)
+    assert isinstance(cb.slot_repr(), str)
+    assert cb.dereference() is not None
+
+    cb.cb((2,))
+    await asyncio.sleep(0.01)
```

Comment:
similar to your comment above... i suspect this is flaky.  Should we have an API to wait until the backend has no items in the queue?

Reply:
That's a possibility, but a simpler approach could be to set an event in the mock, and wait for it (with a timeout)?